experimental/list_formatter/src/deduplicating_array.rs

```diff
@@ -0,0 +1,111 @@
+use core::{mem, ops::Index, ptr};
+use icu_provider::yoke::*;
+
+#[derive(Debug)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Deserialize, serde::Serialize)
+)]
+// TODO: Make this generic over the 12. Serde currently can't derive
+// if const generics are present: https://github.com/serde-rs/serde/issues/1937
+pub struct DeduplicatingArray<T>([Dedupe<T>; 12]);
+
+#[derive(Debug, PartialEq)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Deserialize, serde::Serialize)
+)]
+#[serde(untagged)]
+enum Dedupe<T> {
```

Comment:
It is technically possible for the custom derive to handle this case: when I implemented generic handling I explicitly punted this for later, but it's not too tricky to make happen.

I might need to add an attribute like:

```rust
#[derive(Yokeable, ZCF)]
enum Dedupe<#[yoke(yokeable_param) D> {..}
```

thoughts, @sffc ? I can add this if folks want.

Reply:
Seems useful, but let's align on the architecture of the class first.

Reply:
I don't think this type needs to be `Yokeable` though? It's an implementation detail, and `DeduplicatingArray` can be yoked and zcf'ed without it.

Reply:
Well, if this was Yokeable then DeduplicatingArray could use `#[derive(Yokeable)]` instead of having to do the manual thing